report/2.Analysis/EmbeddedPlatforms.tex

```diff
@@ -0,0 +1,76 @@
+
+The Lego Mindstorm platform, comes with a graphical drag and drop interface.
+However, various different compilers for many different languages exist.
+As such the choice between different generations, does not come down to language preferences.
+
+On the 5th semester of Software, each group is supplied with a lego kit, consisting of both a Lego Mindstorms NXT 2.0 kit and a Lego Mindstorms EV3 kit, aswell as an assortment of sensors, motors and Legos.
+On top of this, it is also possible to order other platforms and sensors.
+
+\subsubsection{Lego Mindstorms NXT 2.0}
+The Lego Mindstorms NXT 2.0 (NXT) is the second generation Lego Mindstorms kit.
+The NXT Intelligent Brick is a micro computer based on the 32-bit 48 MHz ARM7 microprocessor, with 256 kbytes of flash memory and 64 kbytes of RAM\cite{nxt2userguide}\cite{nxt2ev3compare}.
+It includes a variety of sensors, such as touch, sound, light and ultrasonic (vision) sensors.
+On top of this, the consumer kit also includes servo motors and lamps.
+
+A potential major disadvantage of the NXT 2.0, is that the platform is rather old.
+This means that problems could arise, especially in regards to compatability with newer operating systems.
+Further testing will be done and this will be further elaborated on in a later section.
+
+\subsubsection{Lego Mindstorms EV3}
+The  Lego Mindstorms EV3 (EV3) is the third generation of the Lego Mindstorms line.
+The EV3 mainly improves on the specifications of the second generation, adding a significantly more powerful 300MHz ARM9 processor, running Linux, 16 MB of flash memory, which is extendable with a microSDHC card and 64MB of RAM\cite{ev3userguide}.
+On top of this, it also adds remote control and WiFi capabilities, as well as more recently updated compilers.
+
+It is also fully backwards-compatible with the NXT sensors\cite{ev3nxtcompatability}.
+
+%Don't know how relevant this will actually be.
+\subsection{Single-board computers}
+Besides the Lego Mindstorms platform, which is the recommended platform for this semester, there is also the option of using single-board computers, such as the Raspberry Pi or the Arduino.
+Both of are a bit more akin to general purpose computers, designed for small electronics projects and both platforms include different models, with some being more powerful than others or having different features that further set them apart.
+
+\subsubsection{Arduino}
+The Arduino is a single-board computer platform, designed for small electronics projects.
+The Arduino platform is an electronics eco-system, comprised of different boards, that share the trait of being easily programmable in the included language, a subset of \texttt{C++}.
+Arduino boards typically includes a processor, some RAM and flash memory, as well as programmable GPIO pins\footnote{General Purpose Input Output}.
+
+The primary advantage of the Arduino platform is that it is easy to work with, and very modular.
```

Comment:
enten fjern modulær - for jeg ved ikke hvad du mener her - eller skriv hvad du mener.